threads/server-tcp.c

```diff
@@ -0,0 +1,154 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <unistd.h>
+#include <inttypes.h>
+#include <string.h>
+#include <pthread.h>
+
+/****************************************
+        Author: Tim Wood
+        Co-Sub-Authors: Eric Armbrust, Neel Shah, Phil Lopreiato
+        with a little help from
+        http://beej.us/guide/bgnet/
+****************************************/
+
+#define BACKLOG 10     // how many pending connections queue will hold
+
+/*
+ * Max number of concurrent threads.
+ * Don't have more than this or bad things *will* happen :c
+ */
+#define MAX_CLIENTS 512
+
+/****************************************
+ * Func passed to pthread_create that handles
+ * the print off post-connection. If more threads
+ * than MAX_CLIENTS call this function, it esplodes
+ * ...don't do it. Serioudsly
+ ****************************************/
```

Comment:
let's be more precise: I predict that serious explosions can happen with a concurrency level of 2.  What if one really really slow client connects, then while it is still sending you data 512 other fast but sequential clients connect?

You don't have to fix this, but let's have messages that make it clear what cases will cause the doom.